spec/dock-spec.js

```diff
@@ -62,6 +62,74 @@ describe('Dock', () => {
     })
   })
 
+  describe('activating the next pane', () => {
+    describe('when the dock has more than one pane', () => {
+      it('activates the next pane', () => {
+        jasmine.attachToDOM(atom.workspace.getElement())
+        const dock = atom.workspace.getLeftDock()
+        const pane1 = dock.getPanes()[0]
+        const pane2 = pane1.splitRight()
+        const pane3 = pane2.splitRight()
+        pane2.activate()
+        expect(pane1.isActive()).toBe(false)
+        expect(pane2.isActive()).toBe(true)
+        expect(pane3.isActive()).toBe(false)
+
+        dock.activateNextPane()
+        expect(pane1.isActive()).toBe(false)
+        expect(pane2.isActive()).toBe(false)
+        expect(pane3.isActive()).toBe(true)
+      })
+    })
+
+    describe('when the dock has only one pane', () => {
+      it('leaves the current pane active', () => {
+        jasmine.attachToDOM(atom.workspace.getElement())
+        const dock = atom.workspace.getLeftDock()
+
+        expect(dock.getPanes().length).toBe(1)
+        const pane = dock.getPanes()[0]
+        expect(pane.isActive()).toBe(true)
+        dock.activateNextPane()
+        expect(pane.isActive()).toBe(true)
+      })
+    })
```

Comment:
@lee-dohm: In addition to the scenarios above where a dock has one pane or multiple panes, you and I discussed adding a test for the scenario where the dock has zero panes. I haven't been able to set up that scenario in a test. I thought something like this would work:

```js
describe('when the dock has no panes', () => {
  it('has no effect', async () => {
    jasmine.attachToDOM(atom.workspace.getElement())
    const dock = atom.workspace.getLeftDock()
    await dock.destroyActivePane()

    expect(dock.getPanes().length).toBe(0)
    dock.activateNextPane()
    expect(dock.getPanes().length).toBe(0)
  })
})
```

But upon closer inspect, it looks like Atom [prevents destruction of the last pane in a container](https://github.com/atom/atom/blob/ae9f24602d80cb15fe0edc15c1aae37d44ee786c/src/pane.coffee#L770-L771). You mentioned that you were able to remove the last pane in a dock. Can you offer any guidance for how I could reproduce that scenario?